widget/entry.go

```diff
@@ -1039,8 +1040,12 @@ func (r *entryRenderer) Destroy() {
 }
 
 func (r *entryRenderer) Layout(size fyne.Size) {
-	r.line.Resize(fyne.NewSize(size.Width, theme.Padding()))
-	r.line.Move(fyne.NewPos(0, size.Height-theme.Padding()))
+	// Used to be ready to support OutlineBorder later in future.
+	remBoxBorder := r.box.StrokeWidth / 2
```

Comment:
I didn't realise this before, but `remBoxBorder` will always be 0.
Did you mean to set it as `theme.EntryUnderlineSize()`?

Reply:
Yes, it is 0, because we are not using border, but if we add border in the future (if that is the case), it won't be 0 anymore

Reply:
Ah I misunderstood. There is no point having code in there that does nothing.
I thought you were adjusting it so that if border were added later it would not change the sizes.

Reply:
> I thought you were adjusting it so that if border were added later it would not change the sizes.

I am doing this. Now I am lost.
canvas.Rectangle size only applies to its content and half-border, the other half is outside its size. If later we add border and we do not have that line, we have this look:

<img width="456" alt="Captura de Pantalla 2021-01-11 a la(s) 12 36 08" src="https://user-images.githubusercontent.com/12239342/104217891-f4986900-5409-11eb-9507-206b242e4d90.png">

But if we keep that line, the entry will look good:

<img width="456" alt="Captura de Pantalla 2021-01-11 a la(s) 12 36 56" src="https://user-images.githubusercontent.com/12239342/104217928-0417b200-540a-11eb-84c7-4a225bd0936e.png">

Reply:
If box.StrokeWith is currently 0 then this code has no impact and the background reaches the edges I think.
If you then set the strokewidth higher it covers the part of the entry, changing the usable space.
I had thought this was to ensure we didn't change that space and/or the outside size.

If the alteration is to move the rectangle in from the edge, and have no impact on anything else, then the code will do nothing until a future release when we add outlines - so it's just confusing to leave it in at this time.

Reply:
> the code will do nothing until a future release when we add outlines

In fact, that was the idea, that's why I wrote "// Used to be ready to support OutlineBorder later in future".

So I will delete it now to avoid confusion.

Reply:
@andydotxyz I have just removed the unused code.